src/plans/D3MAaveDaiPlan.sol

```diff
@@ -0,0 +1,213 @@
+        require(variableDebt_ != address(0), "D3MAaveDaiPlan/invalid-variableDebt");
+        require(interestStrategy_ != address(0), "D3MAaveDaiPlan/invalid-interestStrategy");
+
+        adai = adai_;
+        stableDebt = TokenLike(stableDebt_);
+        variableDebt = TokenLike(variableDebt_);
+        interestStrategy = InterestRateStrategyLike(interestStrategy_);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Math ---
+    uint256 constant RAY  = 10 ** 27;
+
+    function _add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x + y) >= x, "D3MAaveDaiPlan/overflow");
+    }
+    function _sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x, "D3MAaveDaiPlan/underflow");
+    }
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "D3MAaveDaiPlan/overflow");
+    }
+    function _rmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, y) / RAY;
+    }
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, RAY) / y;
+    }
+
+    // --- Admin ---
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "bar") {
+            require(data <= _maxBar(), "D3MAaveDaiPlan/above-max-interest");
+
+            bar = data;
+        } else revert("D3MAaveDaiPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        if (what == "interestStrategy") interestStrategy = InterestRateStrategyLike(data);
+        else revert("D3MAaveDaiPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function _maxBar() internal view returns (uint256) {
+        return interestStrategy.getMaxVariableBorrowRate();
+    }
+
+    // --- Automated Rate targeting ---
+    function calculateTargetSupply(uint256 targetInterestRate) external view returns (uint256) {
+        uint256 totalDebt = _add(stableDebt.totalSupply(), variableDebt.totalSupply());
+        return _calculateTargetSupply(targetInterestRate, totalDebt);
+    }
+
+    function _calculateTargetSupply(uint256 targetInterestRate, uint256 totalDebt) internal view returns (uint256) {
+        uint256 base = interestStrategy.baseVariableBorrowRate();
+        require(targetInterestRate > base, "D3MAaveDaiPlan/target-interest-base");
+        require(targetInterestRate <= _maxBar(), "D3MAaveDaiPlan/above-max-interest");
+
+        // Do inverse calculation of interestStrategy
+        uint256 variableRateSlope1 = interestStrategy.variableRateSlope1();
+        uint256 targetUtil;
+        if (targetInterestRate > _add(base, variableRateSlope1)) {
+            // Excess interest rate
+            uint256 r = targetInterestRate - base - variableRateSlope1;
```

Comment:
Can be unchecked in 0.8.13

Reply:
Looks like this repo is still on 0.6.X though?